### DISCUSSION TITLE
Cleanup in Joomla 3 prior to Joomla 4

### DISCUSSION THREAD

**Discussion by rgtr** [joomla/joomla-cms#32155] [Q&A] [answered]
Hi
I have eight joomla web sites, one started in J1.5. It is one of the five that have converted ok.

The others all rely on packages that are not compliant - but are expected to be.
Cleaning up is all important, as is making sure that the packages one uses are compliant.

A list of what is and isn't compliant is really needed - just try and install a package in an empty j4 and see if it is possible.

The next thing is the garbage that has been left over after upgrades etc etc.
plg_finder_weblinks and plg_content_geshi from joomla etc. who should remove these ... 

attached my analysis as it currently stands. Components still to be added. I am also listing all the tables using phpmyadmin.

[Joomla-4-structure_02.pdf](https://github.com/joomla/joomla-cms/files/5866061/Joomla-4-structure_02.pdf)

Richard

**Comment by wilsonge** [comment, marked as answer]
> A list of what is and isn't compliant is really needed - just try and install a package in an empty j4 and see if it is possible.

This is a feature in Joomla 3.10 that will be provided (you can try the nightly for that already). Weblinks I believe it is intended to be left in place - we deliberately didn't remove any Weblinks extensions from 4.x - because we have no clue if people are using them - but there should be an update server still in place for all those extensions

**Reply by bembelimen** [reply]
And here is the link for the test installation: https://developer.joomla.org/nightly-builds.html#joomla-310
